api/models/User.js

```diff
@@ -34,14 +35,13 @@ var phoneValidator = [
 ];
 
 var UserSchema = new Schema({
```

Comment:
Important stuff going on here:

1) facebookId is the unique ID of a facebook user
2) "index: { unique: true, sparse: true } }" means "if this field is not null it must be unique"